ql/src/java/org/apache/hadoop/hive/ql/exec/ReduceSinkOperator.java

```diff
@@ -304,9 +294,9 @@ public void process(Object row, int tag) throws HiveException {
             conf.getOutputKeyColumnNames(), numDistributionKeys, rowInspector);
         valueObjectInspector = initEvaluatorsAndReturnStruct(valueEval,
             conf.getOutputValueColumnNames(), rowInspector);
-        partitionObjectInspectors = initEvaluators(partitionEval, rowInspector);
+        partitionHashFunc = conf.getPartitionFunction(initEvaluators(partitionEval, rowInspector));
```

Comment:
could we please extract `initEvaluators` into local var (i.e. `ObjectInspector[] partitionObjectInspectors`)? and do the same for `bucketObjectInspectors`

Reply:
Done
https://github.com/apache/hive/pull/5409/commits/072ff8e6094d08013e935e9b7ad12764af0c52f4